addons/udes_stock_picking_batch/models/stock_picking.py

```diff
@@ -52,12 +52,20 @@ def _trigger_batch_state_recompute(self):
 
     def action_assign(self):
         """
-        Recompute batch state. In theory this is not necessary
+        Ensure the batch state is recomputed. In theory this is not necessary
         but the constraint on state has not proven to work correctly.
         """
         super().action_assign()
         self.batch_id._compute_state()
 
+    def _action_done(self):
+        """
+        Ensure the batch state is recomputed. In theory this is not necessary
+        but the constraint on state has not proven to work correctly.
+        """
+        super()._action_done()
+        self.batch_id._compute_state()
+
```

Comment:
Add a line: `return res` at the end. The reason is because the core function is returning True at the end and not sure if somewhere is needed the value that returns. I think its a good practice to return it even when the code method doesnt return anything

Reply:
Thanks, I overlooked this.